xapian-core/include/xapian/weight.h

```diff
@@ -315,6 +316,25 @@ class XAPIAN_VISIBILITY_DEFAULT Weight {
 	return stats_needed & UNIQUE_TERMS;
     }
 
+    /** @private @internal Return the appropriate weighting scheme object.
```

Comment:
You've marked the new `create()` method as `@private @internal` which means it won't appear in the collated API documentation.